atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -771,13 +778,17 @@ private void extractLocalWritesForRow(
     @Override
     @Idempotent
     public Map<Cell, byte[]> get(TableReference tableRef, Set<Cell> cells) {
-        return AtlasFutures.getUnchecked(
-                getInternal("get", tableRef, cells, immediateKeyValueService, immediateTransactionService));
+        return cache.get(
+                tableRef,
+                cells,
+                (table, uncached) ->
+                        getInternal("get", tableRef, cells, immediateKeyValueService, immediateTransactionService));
     }
 
     @Override
     @Idempotent
     public ListenableFuture<Map<Cell, byte[]>> getAsync(TableReference tableRef, Set<Cell> cells) {
+        // todo(gmaretic): make this use cache as well?
         return getInternal("getAsync", tableRef, cells, keyValueService, defaultTransactionService);
```

Comment:
the cache's `get` method is literally just a `AtlasFutures.getUnchecked` on the `getInternal` - not against just renaming and exposing that so that we have async as well.

Reply:
That's probably outiside the scope here

Reply:
https://github.com/palantir/atlasdb/pull/5413 implements `getAsync`, but we can merge that in the next PR (probably too painful to do _yet another_ rebase here :P)